exam1.cpp

```diff
@@ -0,0 +1,369 @@
+    }
+
+    string getCodigo(){return codigoAsiento;}
+    bool estaDisponible(){return disponible;}
+
+    void ocupar(){disponible = false;}
+    void liberar(){ disponible =true;}
+
+
+};
+
+class Pasajero
+{
+    private:
+    string ci;
+    string nombre;
+    string apellido;
+    Asiento* asientoAsignado;
+
+    public:
+    Pasajero(string _ci , string _nombre , string _apellido)
+    {
+        this ->ci = _ci;
+        this ->nombre = _nombre;
+        this -> apellido = _apellido;
+        this ->asientoAsignado = nullptr;
+    }
+
+    string getCI(){return ci;}
+    string getNombre(){return nombre;}
+    string getApellido(){return apellido;}
+    Asiento *getAsiento(){return asientoAsignado;}
+
+    void asignarAsiento(Asiento* asiento)
+    {
+        this ->asientoAsignado = asiento;
+    }
+
+    void mostrar()
+    {
+        cout << " CI " << ci <<" / " << apellido <<", "<<nombre;
+        if (asientoAsignado != nullptr)
+        {
+            cout << " / asiento " << asientoAsignado->getCodigo();
+        }
+        cout << endl;
+    }
+
+};
+
+bool compararAsientoPorCodigo(string codigo , Asiento* asiento)
+{
+    return asiento ->getCodigo() == codigo;
+}
+
+bool compararPasajePorCI(string ci , Pasajero* pasajero)
+{
+    return pasajero ->getCI() == ci ;
+}
+
+bool compararPasajeroPorApellidos(Pasajero*a , Pasajero* b)
+{
+    return a->getApellido() > b ->getApellido();
+}
+
+class Vuelo
+{
+    private:
+    string codigoVuelo;
+    int horaPartida;
+    int horaLlegadaEsperada;
+    int horaLlegadaActual;
+    ArrayGenerica<Asiento>* asientos;
+    ArrayGenerica<Pasajero>* pasajeros;
+    int totalAsientos;
+
+    public:
+
+    Vuelo(string _codigo,int _horaPartida, int _horaLlegadaEsperada, int llegadaActual, int filas , int columnas)
+    {
+        this ->codigoVuelo = _codigo;
+        this ->horaPartida = _horaPartida;
+        this->horaLlegadaEsperada=_horaLlegadaEsperada;
+        this->horaLlegadaActual=llegadaActual;
+        this->totalAsientos= filas * columnas;
+        this ->asientos = new ArrayGenerica<Asiento>(totalAsientos);
+        this->pasajeros = new ArrayGenerica<Pasajero>(totalAsientos);
+    
+        char letras[15] = {'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L'};
+            for (int fila = 1; fila <= filas; fila++) {
+                for (int col = 0; col < columnas; col++) {
+                string codigoAsiento = string(1, letras[col]) + to_string(fila);
+                asientos->agregar(new Asiento(codigoAsiento));
+            }
+        }
+    }
+
+    string getCodigo()
+    {
+        return codigoVuelo;
+    }
+
+    int getHoraPartida()
+    {
+        return horaPartida;
+    }
+
+    int getHoraLlegadaEsperada()
+    {
+        return horaLlegadaEsperada;
+    }
+    int getHoraLlegadaActual()
+    {
+        return horaLlegadaActual;
+    }
+
+    void registrarPasajeros(string ci , string nombre , string apellido, string codigoAsiento)
```

Comment:
Mejor si es un objeto el pasajero
```suggestion
    void registrarPasajeroYAsignarAsiento(new Pasajero* nuevoPasajero , string codigoAsiento)
```